htdocs/core/tpl/massactions_pre.tpl.php

```diff
@@ -73,7 +73,8 @@
 	$formquestion = array();
 	if (!empty($categ_types)) {
 		foreach ($categ_types as $categ_type) {
-			$categ_arbo_tmp = $form->select_all_categories($categ_type['code'], null, 'parent', null, null, 2);
+			//$categ_arbo_tmp = $form->select_all_categories($categ_type['code'], null, 'parent', null, null, 2); //does not show the category hierarchy
+			$categ_arbo_tmp = $form->select_all_categories($categ_type['code'], '', 'parent', 64, 0, 1); //show the category hierarchy
```

Comment:
Using the new param 2 instead of 1 seems to return a more completearray
Which data do you expect to have with param 1 you don't have with param 2 ?